repos/spack_repo/builtin/packages/py_hatch/package.py

```diff
@@ -37,5 +39,7 @@ class PyHatch(PythonPackage):
     depends_on("py-tomlkit@0.11.1:", type=("build", "run"))
     depends_on("py-userpath@1.7:1.7", type=("build", "run"))
```

Comment:
While you are at it can you fix the dependency restriction here?
According to the `pyproject.toml` it is `userpath~=1.7`
```suggestion
    depends_on("py-userpath@1.7:1", type=("build", "run"))
```